### PR TITLE
Add via keymap to plaid

### PR DESCRIPTION
**Pull request by hsgw**
## Description

Add via keymap to plaid.   

Default PID/VID are defined in post_config.h.

## Note
And this will not work until #8634 is merged.

PR to via @yiancar 
https://github.com/the-via/keyboards/pull/76

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by hsgw** [comment]
I'm sure travis-ci errors are due to community layout for other keyboard.

**Comment by yiancar** [comment]
I think what he is trying to here is maintain the legally free VID PIDs when not necessary and use some unique but maybe illegal ones for via which is a necessity

**Comment by hsgw** [comment]
> Not a fan of the use of post_config.
> 
> We should just change the IDs in `keyboards/plaid/config.h` or undef and then def in `keyboards/plaid/keymaps/via/config.h` ( though I don't really see the point in this direction either).

I know what you mean. And I think both way is fine.
But why shouldn't use `post_config.h` ?

**Comment by zvecr** [comment]
I would rather not introduce a new mechanism into this problem and fragment the code base further. We have been changing PID/VIDs at the keyboard level and we should continue to do so.

**Comment by hsgw** [comment]
> I would rather not introduce a new mechanism into this problem and fragment the code base further.
Certainly it might be too much to create a new file for trivial change.  I'll change this.

> changing PID/VIDs at keymap level
I know this is not clean way.
But I would like to keep using safe VID/PID pair for default keymap.

File: keyboards/plaid/keymaps/via/keymap.c [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+
+#define LOWER MO(1)
+#define RAISE MO(2)
```
Comment (zvecr) [inline keyboards/plaid/keymaps/via/keymap.c:20]:
```suggestion
#define LOWER  FN_MO13
#define RAISE  FN_MO23
```

File: keyboards/plaid/keymaps/via/keymap.c [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+    _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______
+)
+
+};
+
+uint32_t layer_state_set_user(uint32_t state) {
+  return update_tri_layer_state(state, 1, 2, 3);
+}
+
```
Comment (zvecr) [inline keyboards/plaid/keymaps/via/keymap.c:101]:
```suggestion

```

**Comment by Wilba6582** [comment]
> Not a fan of the use of post_config.
> 
> We should just change the IDs in `keyboards/plaid/config.h` or undef and then def in `keyboards/plaid/keymaps/via/config.h` ( though I don't really see the point in this direction either).

Agreed.

**Comment by hsgw** [comment]
I changed.
VID/PID are undef and re-define in keymap/via/config.h.

**Comment by zvecr** [comment]
Thanks!

**Pull request merged**